java-squid/src/main/java/org/sonar/java/collections/AVLTree.java

```diff
@@ -0,0 +1,388 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012 SonarSource
+ * sonarqube@googlegroups.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.java.collections;
+
+import com.google.common.base.Preconditions;
+
+import javax.annotation.Nullable;
+
+import java.util.Comparator;
+import java.util.Objects;
+
+/**
+ * AVL Tree.
```

Comment:
I would prefer a slightly extended description, with for instance the addition of ``(self-balancing binary search tree)`` to at least give an idea of what it could be without going on wikipedia.